src/styles/main.scss

```
@@ -1,3 +1,71 @@
+@use 'sass:list';
+
 body {
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
 }
+
+.calendar {
+  --cell-size: 100px;
+  --gap-size: 1px;
+  --padding: 10px;
+
+  $days: mon, tue, wed, thu, fri, sat, sun;
+
+  @each $day in $days {
+    $i: list.index($days, $day);
+      &--start-day-#{$day} {
+      .calendar__day:first-child {
+        margin-left: calc((var(--cell-size) + var(--gap-size)) * #{$i - 1});
+      }
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} {
+      .calendar__day:nth-child(n + #{$i + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  width: calc(7 * var(--cell-size) + 6 * var(--gap-size) + 2 * var(--padding));
+  display: flex;
+  gap: var(--gap-size);
+  flex-wrap: wrap;
+  padding: var(--padding);
+  box-sizing: border-box;
+
+  &__day {
+    box-sizing: border-box;
+    display: inline-block;
+    position: relative;
+    border: 1px solid #000;
+    background-color: #eee;
+    width: 100px;
+    height: 100px;
+    margin: 0;
+    cursor: pointer;
+    animation-duration: 0.5s;
```

```suggestion
    transition-duration: 0.5s;
```
Pay attention to the difference between transition and animation
https://blog.hubspot.com/website/css-transition-vs-animation